zerver/lib/export.py

```diff
@@ -1182,26 +1189,34 @@ def _get_exported_s3_record(
     if processing_emoji:
         record['file_name'] = os.path.basename(key.name)
 
-    # A few early avatars don't have 'realm_id' on the object; fix their metadata
-    user_profile = get_user_profile_by_id(record['user_profile_id'])
-    if 'realm_id' not in record:
-        record['realm_id'] = user_profile.realm_id
-    record['user_profile_email'] = user_profile.email
+    if "user_profile_id" in record:
+        user_profile = get_user_profile_by_id(record['user_profile_id'])
+        record['user_profile_email'] = user_profile.email
 
-    # Fix the record ids
-    record['user_profile_id'] = int(record['user_profile_id'])
-    record['realm_id'] = int(record['realm_id'])
+        # Fix the record ids
+        record['user_profile_id'] = int(record['user_profile_id'])
 
-    return record
+        # A few early avatars don't have 'realm_id' on the object; fix their metadata
+        if 'realm_id' not in record:
+            record['realm_id'] = user_profile.realm_id
+    else:
+        # There are some rare cases in which 'user_profile_id' may not be present
+        # in S3 metadata. Eg: Exporting an organization which was created
+        # initially from a local export won't have the "user_profile_id" metadata
+        # set for realm_icons and realm_logos.
+        pass
+
+    if 'realm_id' in record:
+        record['realm_id'] = int(record['realm_id'])
+    else:
+        raise Exception("Missing realm_id")
 
-def _save_s3_object_to_file(
-        key: Key,
-        output_dir: str,
-        processing_avatars: bool,
-        processing_emoji: bool) -> None:
+    return record
 
+def _save_s3_object_to_file(key: Key, output_dir: str, processing_avatars: bool,
+                            processing_emoji: bool, processing_realm_icon_and_logo: bool) -> None:
     # Helper function for export_files_from_s3
-    if processing_avatars or processing_emoji:
+    if processing_avatars or processing_emoji or processing_realm_icon_and_logo:
```

Comment:
If I'm reading this function correctly, it doesn't really need all of these arguments; it could just take a `check_key` boolean, or perhaps better, we could just move the `len(fields)` check into the caller.